mydburl_test.go

```diff
@@ -97,6 +98,45 @@ func TestOpen(t *testing.T) {
 	}
 }
 
+func TestRegisterTlsConfig(t *testing.T) {
+	port, ca, cert, key := createMySQLContainer(t)
+	tests := []struct {
+		dsn     string
+		wantErr bool
+	}{
+		{fmt.Sprintf("mysql://%s:%s@localhost:%s/%s?parseTime=true", mysqlUser, mysqlPassword, port, mysqlDatabase), true},
+		{fmt.Sprintf("mysql://%s:%s@localhost:%s/%s?parseTime=true&sslCa=%s", mysqlUser, mysqlPassword, port, mysqlDatabase, ca), false},
+		{fmt.Sprintf("mysql://%s:%s@localhost:%s/%s?parseTime=true&sslCa=%s&sslCert=%s&sslKey=%s", mysqlUser, mysqlPassword, port, mysqlDatabase, ca, cert, key), false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.dsn, func(t *testing.T) {
+			u, err := mydburl.Parse(tt.dsn)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if err := u.RegisterTlsConfig("test"); err != nil {
+				if tt.wantErr {
+					return
+				}
+				t.Error(err)
+			}
+			if strings.Contains(u.String(), "sslCa") {
+				t.Error("sslCa should be removed")
+			}
+			if !strings.Contains(u.String(), "tls") {
+				t.Error("tls should be added")
+			}
+			db, err := sql.Open(u.Driver, u.DSN)
```

Comment:
With this change, we get the info we need for `sql.Open` without having to use `mydburl.Open`.